### PR TITLE
feat: Trigger chat-visitor-registered on 'new chat' after closing

### PR DESCRIPTION
**Pull request by MartinSchoeler**
This will allow the `after-guest-registration` trigger to run when starting a new chat after finishing a chat when the `clear guest data when chat ended` setting is set to false

[SUP-483]

[SUP-483]: https://rocketchat.atlassian.net/browse/SUP-483?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

**Review by MarcosSpessatto — CHANGES_REQUESTED** [review]
Can we add a changeset please?

**Review by MarcosSpessatto — APPROVED** [review]

**Review by KevLehman — DISMISSED** [review]

File: apps/meteor/tests/e2e/omnichannel/omnichannel-triggers.spec.ts [hunk 1]
```diff
@@ -166,3 +166,79 @@ test.describe.serial('OC - Livechat Triggers', () => {
...
+			api.post('/settings/Livechat_clear_local_storage_when_chat_ended', { value: false }),
+		]);
+		await agent.page.close();
+	});
+
+	test('OC - Livechat New Chat Triggers - After Registration', async () => {
+		await test.step('expect trigger message after registration', async () => {
+			await expect(poLiveChat.txtChatMessage(triggerMessage)).toBeVisible();
+		});
+	});
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-triggers.spec.ts:237]:
```suggestion
	test.describe('OC - Livechat New Chat Triggers - After Registration', async () => {
		await test('expect trigger message after registration', async () => {
			await expect(poLiveChat.txtChatMessage(triggerMessage)).toBeVisible();
		});
	});
```

File: apps/meteor/tests/e2e/omnichannel/omnichannel-triggers.spec.ts [hunk 2]
```diff
@@ -166,3 +166,79 @@ test.describe.serial('OC - Livechat Triggers', () => {
...
+
+	test('OC - Livechat New Chat Triggers - After Registration, clear Local storage', async () => {
+		await test.step('expect trigger message after registration', async () => {
+			await expect(poLiveChat.txtChatMessage(triggerMessage)).toBeVisible();
+		});
+	});
```
Comment (ggazzo) [inline apps/meteor/tests/e2e/omnichannel/omnichannel-triggers.spec.ts:243]:
```suggestion
	test.describe('OC - Livechat New Chat Triggers - After Registration, clear Local storage', async () => {
		test.beforeEach(async ({ page, api }, testInfo) => {
			await api.post('/settings/Livechat_clear_local_storage_when_chat_ended', { value: true });
		});
	
		await test('expect trigger message after registration', async () => {
			await poLiveChat.page.reload();
			await poLiveChat.sendMessage(newUser, false);
			await expect(poLiveChat.txtChatMessage(triggerMessage)).toBeVisible();
		});
	});
```

**Review by aleksandernsilva — DISMISSED** [review]

**Comment by ggazzo** [comment]
as expected both tests are failing before the fix:
```
✘  60 omnichannel/omnichannel-triggers.spec.ts:229:13 › OC - Livechat New Chat Triggers › OC - Livechat New Chat Triggers - After Registration › expect trigger message after registration (5.0s)
Sending test result to Rocket.Chat: {"name":"expect trigger message after registration","status":"failed","duration":5028,"branch":"refs/pull/31718/merge","draft":false,"run":17995}
  ✘  61 omnichannel/omnichannel-triggers.spec.ts:248:13 › OC - Livechat New Chat Triggers › OC - Livechat New Chat Triggers - After Registration, clear Local storage › expect trigger message after registration (5.3s)
```

**Review by ggazzo — APPROVED** [review]

**Pull request merged**